prerendering.bs

```diff
@@ -478,172 +459,146 @@ Every {{Document}} has an <dfn for="Document">activation start time</dfn>, which
-      * |browsingContext|'s [=active document=]'s [=Document/prerendering browsing contexts map=][|request|'s [=request/URL=]] [=map/exists=] and is not [=prerendering browsing context/empty=]
-      * The result of calling [=should navigation request of type be blocked by content security policy?=] given |request| and <var ignore>navigationType</var> is "`Allowed`".
+      * |navigable| is not a [=top-level traversable=];
+      * |navigable| is a [=prerendering traversable=];
+      * |navigable| is a [fenced frame](https://github.com/shivanigithub/fenced-frame);
+         <p class="issue">The concept of a fenced frame navigable is not yet defined but we need to link to it once it exists.</p>
+      * |cspNavigationType| is not "`other`"; or
+      * |documentResource| is not null
 
-  1. Otherwise, return false.
+    then return null.
 
+  1. Return |navigable|'s [=navigable/active document=]'s [=Document/prerendering traversables map=][|url|], if it [=map/exists=]; otherwise return null.
 </div>
 
-Patch the [=navigate=] algorithm to allow the [=prerendering browsing context/activate|activation=] of a [=prerendering browsing context=] in place of a normal navigation as follows:
+Patch the [=navigate=] algorithm to allow the [=prerendering traversable/activate|activation=] of a [=prerendering traversable=] in place of a normal navigation as follows:
 
 <div algorithm="navigate activate patch">
-  In [=navigate=], append the following steps after the fragment navigation handling (currently step 8):
+  In [=navigate=], insert the following steps as the first ones after we go [=in parallel=]:
 
-  1. If |resource| is a [=request=] and we [=can activate a prerender=] given |browsingContext|, |historyHandling|, <var ignore>navigationType</var>, and |resource|, then:
+  1. Let |matchingPrerenderedNavigable| be the result of [=getting the matching prerendering navigable=] given |navigable|, <var ignore>url</var>, <var ignore>cspNavigationType</var>, and <var ignore>documentResource</var>.
 
-    1. Let |successorBC| be |browsingContext|'s [=active document=]'s [=Document/prerendering browsing contexts map=][|resource|'s [=request/URL=]].
+  1. If |matchingPrerenderedNavigable| is not null, then:
 
-    1. Wait until all navigation attempts of |successorBC| have been [=navigate/mature|matured=].
+    1. Wait until either |matchingPrerenderedNavigable|'s [=navigable/active document=]'s [=Document/is initial about:blank=] is false, or |matchingPrerenderedNavigable| is [=destroy a top-level traversable|destroyed=].
 
-    1. If |successorBC| has not been [=discard|discarded=], then:
+    1. If |matchingPrerenderedNavigable| was not destroyed, then:
 
-      1. [=prerendering browsing context/Activate=] |successorBC| in place of |browsingContext| given |historyHandling|.
+      1. [=prerendering traversable/Activate=] |matchingPrerenderedNavigable| in place of |navigable| given <var ignore>historyHandling</var>.
```

Comment:
Good catch. I think this is leftover from before I made activate happen entirely in the "in parallel" section.

I think the right behavior here is pretty unclear because of the WebDriver BiDi stuff. Also, UA-initiated prerendering is a bit more annoying. But threading the navigation ID through seems at least a bit more correct. I'll open a WebDriver BiDi issue and in the meantime thread things through, even though that'll give somewhat strange behavior.